tools/halide_image.h

```diff
@@ -1,278 +1,696 @@
-// This header defines a simple Image class which wraps a buffer_t. This is
-// useful when interacting with a statically-compiled Halide pipeline emitted by
-// Func::compile_to_file, when you do not want to link your processing program
-// against Halide.h/libHalide.a.
+// A simple Image class which wraps a buffer_t. This is useful
```

Comment:
This comment needs to be rewritten to be stronger and make the following points:
1) This is an enhanced C++ interface on halide_buffer_t (buffer_t).
2) It optionally provides RAII on the underlying storage.
3) It does not "wrap" buffer_t but inherits from it.
4) When using Halide generated code from C++ this is likely the preferred way to declare a halide_buffer_t
5) (Temporary) This is intended to provide a seamless transition from buffer_t to halide_buffer_t for code written against this class.
6) It should document any overhead it incurs above the underlying halide_buffer_t.